sphinx_gallery/directives.py

```diff
@@ -77,3 +80,198 @@ def run(self):
+    """
+    Implements a directive to allow an optional hidpi image.  Meant to be
+    used with the `image_srcset` configuration option.
+
+    e.g.::
+
+        .. image-sg:: /plot_types/basic/images/sphx_glr_bar_001.png
+            :alt: bar
+            :srcset: /plot_types/basic/images/sphx_glr_bar_001.png,
+                     /plot_types/basic/images/sphx_glr_bar_001_2_0x.png 2.0x
+            :class: sphx-glr-single-img
+
+    The resulting html is::
+
+        <img src="sphx_glr_bar_001_hidpi.png"
+            srcset="_images/sphx_glr_bar_001.png,
+                    _images/sphx_glr_bar_001_2_0x.png 2x",
+            alt="bar"
+            class="sphx-glr-single-img" />
+
+    """
+
+    has_content = False
+    required_arguments = 1
+    optional_arguments = 3
+    final_argument_whitespace = False
+    option_spec = {
+        'srcset': directives.unchanged,
+        'class': directives.class_option,
+        'alt': directives.unchanged,
+    }
+
+    def run(self):
+        image_node = imgsgnode()
+
+        imagenm = self.arguments[0]
+        image_node['alt'] = self.options.get('alt', '')
+        image_node['class'] = self.options.get('class', None)
+
+        # we would like uri to be the highest dpi version so that
+        # latex etc will use that.  But for now, lets just make
+        # imagenm
+
+        image_node['uri'] = imagenm
+        image_node['srcset'] = self.options.get('srcset', None)
+
+        return [image_node]
+
+
+def _parse_srcset(st):
+    """ parse st"""
+    entries = st.split(',')
+    srcset = {}
+    for entry in entries:
+        spl = entry.strip().split(' ')
+        if len(spl) == 1:
+            srcset[0] = spl[0]
+        elif len(spl) == 2:
+            mult = spl[1][:-1]
+            srcset[float(mult)] = spl[0]
+    return srcset
+
+
+def visit_imgsg_html(self, node):
+
+    if node['srcset'] is None:
+        self.visit_image(node)
+        return
+
+    imagedir, srcset = _copy_images(self, node)
+
+    # /doc/examples/subd/plot_1.rst
+    docsource = self.document['source']
+    # /doc/
+    # make sure to add the trailing slash:
+    srctop = os.path.join(self.builder.srcdir, '')
+    # examples/subd/plot_1.rst
+    relsource = os.path.relpath(docsource, srctop)
+    # /doc/build/html
+    desttop = os.path.join(self.builder.outdir, '')
+    # /doc/build/html/examples/subd
+    dest = os.path.join(desttop, relsource)
+
+    # ../../_images/
+    imagerel = os.path.relpath(imagedir, os.path.dirname(dest))
+    imagerel = os.path.join(imagerel, '')
+    if '\\' in imagerel:
+        imagerel = imagerel.replace('\\', '/')
+    # make srcset str.  Need to change all the prefixes!
+    srcsetst = ''
+    for mult in srcset:
+        nm = os.path.basename(srcset[mult][1:])
+        # ../../_images/plot_1_2_0x.png
+        relpath = imagerel+nm
+        srcsetst += f'{relpath}'
+        if mult == 0:
+            srcsetst += ', '
+        else:
+            srcsetst += f' {mult:1.1f}x, '
```

Comment:
```suggestion
        if mult != 0:
            srcsetst += f' {mult:1.1f}x'
        srcsetst += ', '
```

Reply:
Thats more elegant, but less clear to me.  I'm a very literal programmer ;-)

Reply:
To me it also matches better with the statement below about removing the trailing comma-space, but agreed it's really six of one half a dozen of the other so is fine as is